ansible_ai_connect/users/serializers.py

```diff
@@ -51,4 +51,20 @@ def get_content(self, user):
                     """
                     break
 
+        if (
+            settings.ANSIBLE_AI_ENABLE_ONE_CLICK_TRIAL
+            and user.is_authenticated
+            and user.is_oidc_user
+            and user.rh_org_has_subscription
+            and not user.org_has_api_key
+        ):
+            markdown_value = f"""
+                Logged in as: {user.username}<br><br>
+                Looks like there is no model set for your organization.<br><br>
+                Start a 90 day free trial with our recommended model,
+                IBM watsonx Code Assistant. This will only apply
+                to you and will not affect your organization.<br><br>
+                <a href = "http://localhost:8000/">Start a trial</vscode-button>
```

Comment:
I suggest a change like this dynamically to get the correct host. This won't provide the right protocol (`http://` vs `https://`), so it defaults on `https://`.
The patch also fix the way you get the `org_has_api_key` value.

```patch
diff --git a/ansible_ai_connect/organizations/models.py b/ansible_ai_connect/organizations/models.py
index f561e937..ae2d60c2 100644
--- a/ansible_ai_connect/organizations/models.py
+++ b/ansible_ai_connect/organizations/models.py
@@ -14,10 +14,13 @@
 
 import logging
 
+from django.apps import apps
 from django.conf import settings
 from django.db import models
 from django.utils.functional import cached_property
 
+from ansible_ai_connect.ai.api.aws.wca_secret_manager import Suffixes
+
 logger = logging.getLogger(__name__)
 
 
@@ -46,6 +49,12 @@ class Organization(models.Model):
             # User should not be blocked if LaunchDarkly fails.
             return True
 
+    @cached_property
+    def has_api_key(self) -> bool:
+        secret_manager = apps.get_app_config("ai").get_wca_secret_manager()
+        org_has_api_key = secret_manager.secret_exists(self.id, Suffixes.API_KEY)
+        return org_has_api_key
+
     def __make_organization_request_to_launchdarkly(self, flag: str) -> bool:
         if not settings.LAUNCHDARKLY_SDK_KEY:
             return False
diff --git a/ansible_ai_connect/users/serializers.py b/ansible_ai_connect/users/serializers.py
index 5d42fa66..6a3f773f 100644
--- a/ansible_ai_connect/users/serializers.py
+++ b/ansible_ai_connect/users/serializers.py
@@ -51,12 +51,14 @@ class MarkdownUserResponseSerializer(serializers.Serializer):
                     """
                     break
 
+        host = self.context["request"].get_host()
+
         if (
             settings.ANSIBLE_AI_ENABLE_ONE_CLICK_TRIAL
             and user.is_authenticated
             and user.is_oidc_user
             and user.rh_org_has_subscription
-            and not user.org_has_api_key
+            and not user.organization.has_api_key
         ):
             markdown_value = f"""
                 Logged in as: {user.username}<br><br>
@@ -64,7 +66,7 @@ class MarkdownUserResponseSerializer(serializers.Serializer):
                 Start a 90 day free trial with our recommended model,
                 IBM watsonx Code Assistant. This will only apply
                 to you and will not affect your organization.<br><br>
-                <a href = "http://localhost:8000/">Start a trial</vscode-button>
+                <a href = "https://{host}/">Start a trial</vscode-button>
             """
 
         return dedent(markdown_value).strip()
diff --git a/ansible_ai_connect/users/views.py b/ansible_ai_connect/users/views.py
index ef13c8aa..33d5f771 100644
--- a/ansible_ai_connect/users/views.py
+++ b/ansible_ai_connect/users/views.py
@@ -143,7 +143,7 @@ class MarkdownCurrentUserView(RetrieveAPIView):
     def retrieve(self, request, *args, **kwargs):
         # User data and use Django to serialise it into a dict
         instance = self.get_object()
-        serializer = self.get_serializer(instance)
+        serializer = self.get_serializer(instance, context={"request": request})
 
         response = serializer.data
 
```

Reply:
Actually, the link should end with `/trial/`, eg: `http://localhost:8000/trial/`